configure.ac

```diff
@@ -95,8 +95,8 @@ PKG_CHECK_EXISTS(
         )]
 )
 
-AC_CHECK_FILE($SOFTOKENDIR/libsoftokn3.so, [],
-    [AC_CHECK_FILE($SOFTOKENDIR/nss/libsoftokn3.so,
+AC_CHECK_FILE($SOFTOKENDIR/libsoftokn3$SHARED_EXT, [],
+    [AC_CHECK_FILE($SOFTOKENDIR/nss/libsoftokn3$SHARED_EXT,
```

Comment:
This is good but not sufficient.

```
$ git grep "\.so"
[..]
configure.ac:AC_CHECK_FILE($SOFTOKENDIR/libsoftokn3.so, [],
configure.ac:    [AC_CHECK_FILE($SOFTOKENDIR/nss/libsoftokn3.so,
configure.ac:    AC_SUBST([P11KITCLIENTPATH], "$P11_MODULE_PATH/p11-kit-client.so")
packaging/pkcs11-provider.spec:%{_libdir}/ossl-modules/pkcs11.so
[..]
tests/openssl.cnf.in:module = @libtoollibs@/pkcs11.so
[..]
tests/setup-softhsm.sh:    /usr/local/lib/softhsm/libsofthsm2.so \
tests/setup-softhsm.sh:    /usr/lib64/pkcs11/libsofthsm2.so \
tests/setup-softhsm.sh:    /usr/lib/pkcs11/libsofthsm2.so \
tests/setup-softhsm.sh:    /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so
tests/setup-softhsm.sh:TMPPDIR="tmp.softhsm"
tests/setup-softhsm.sh:#export PKCS11_PROVIDER_MODULE=/usr/lib64/pkcs11-spy.so
tests/setup-softokn.sh:TMPPDIR="tmp.softokn"
tests/setup-softokn.sh:export PKCS11_PROVIDER_MODULE="${SOFTOKNPATH}/libsoftokn3.so"
```

Reply:
As mentioned in the comment for p11-client-kit.so, some of those are actually .so on macOS as well. I will follow up with a patch to fix those that need fixing, though.